truelearn/learning/_interest_classifier.py

```diff
@@ -1,25 +1,234 @@
+                interest decay function.
+
+        Raises:
+            ValueError:
+                If draw_proba_type is neither "static" nor "dynamic";
+                If decay_func_type is neither "short" nor "long".
+        """
+        self._validate_params(
+            learner_model=learner_model,
+            threshold=threshold,
+            init_skill=init_skill,
+            def_var=def_var,
+            beta=beta,
+            tau=tau,
+            positive_only=positive_only,
+            draw_proba_type=draw_proba_type,
+            draw_proba_static=draw_proba_static,
+            draw_proba_factor=draw_proba_factor,
+            decay_func_type=decay_func_type,
+            decay_func_factor=decay_func_factor,
+        )
+
+        super().__init__(
+            learner_model=learner_model,
+            threshold=threshold,
+            init_skill=init_skill,
+            def_var=def_var,
+            tau=tau,
+            beta=beta,
+            positive_only=positive_only,
+            draw_proba_type=draw_proba_type,
+            draw_proba_static=draw_proba_static,
+            draw_proba_factor=draw_proba_factor,
+        )
+
+        if decay_func_type not in ("short", "long"):
+            raise ValueError(
+                f"The decay_func_type must be either short or long."
+                f" Got {decay_func_type} instead."
+            )
+        self.decay_func_type = decay_func_type
+        self.decay_func_factor = decay_func_factor
+
+    def __get_decay_func(self) -> Callable[[float], float]:
+        """Get decay function based on decay_func_type.
+
+        Returns:
+            A decay function based on given type.
+
+        Notes:
+            Equations from: https://link.springer.com/article/10.1007/s11227-020-03266-2
+        """
+        if self.decay_func_type == "short":
+            return lambda t_delta: min(
+                2 / (1 + math.exp(self.decay_func_factor * t_delta)), 1.0
+            )
+
+        return lambda t_delta: min(math.exp(-self.decay_func_factor * t_delta), 1.0)
+
+    # pylint: disable=too-many-locals
+    def _update_knowledge_representation(self, x: EventModel, y: bool) -> None:
+        if x.event_time is None:
+            raise ValueError(
+                "The event time should not be None when using InterestClassifier."
+            )
+
+        event_time_posix = dt.utcfromtimestamp(x.event_time)
+
+        # make it a list because we need to use it more than one time later
+        # select topic_kc_pairs with default event time = x.event_time
+        learner_topic_kc_pairs = list(
+            select_topic_kc_pairs(
+                self.learner_model,
+                x.knowledge,
+                self.init_skill,
+                self.def_var,
+                x.event_time,
+            )
+        )
+        learner_kcs = map(
+            lambda topic_kc_pair: topic_kc_pair[1], learner_topic_kc_pairs
```

Comment:
Shadows name 'topic_kc_pair' from outer scope 
Maybe call it: learner_topic_kc_pair